mkdocs/nav.py

```diff
@@ -26,6 +29,37 @@ def filename_to_title(filename):
     return utils.filename_to_title(filename)
 
 
+def _path_to_page(path, use_directory_urls, docs_dir, title=None):
+    url = utils.get_url_path(path, use_directory_urls)
+    return Page(title=title, url=url, path=path, docs_dir=docs_dir)
+
+
+def paths_to_pages(pages_config, use_directory_urls, docs_dir):
+
+    for i, entry in enumerate(pages_config):
+
+        if isinstance(entry, utils.text_type):
+            pages_config[i] = _path_to_page(entry, use_directory_urls, docs_dir)
+            continue
+        elif not isinstance(entry, dict):
+            msg = ("Line in 'page' config is of type {0}, dict or string "
```

Comment:
This is the same error we had before, but while I am moving it around we should improve it.